src/compiler/reverse.jl

```diff
@@ -26,13 +26,28 @@ end
 unwrapquote(x) = x
 unwrapquote(x::QuoteNode) = x.value
 
-is_literal_getproperty(ex) =
-  (iscall(ex, Base, :getproperty) || iscall(ex, Core, :getfield) || iscall(ex, Base, :getfield)) &&
-  ex.args[3] isa Union{QuoteNode,Integer}
+is_getproperty(ex) = iscall(ex, Base, :getproperty)
 
 function instrument_getproperty!(ir, v, ex)
-  is_literal_getproperty(ex) ?
-    (ir[v] = xcall(Zygote, :literal_getproperty, ex.args[2], Val(unwrapquote(ex.args[3])))) :
+  if is_getproperty(ex)
+    if ex.args[3] isa Union{QuoteNode,Integer}
+      ir[v] = xcall(Zygote, :literal_getproperty, ex.args[2], Val(unwrapquote(ex.args[3])))
+    else
+      f = insert!(ir, v, :(Val($(ex.args[3]))))
+      ir[v] = xcall(Zygote, :literal_getproperty, ex.args[2], f)
+    end
+  else
+    ex
+  end
+end
+
+is_literal_getfield(ex) =
+  (iscall(ex, Core, :getfield) || iscall(ex, Base, :getfield)) &&
+  ex.args[3] isa Union{QuoteNode,Integer}
+
+function instrument_getfield!(ir, v, ex)
+  is_literal_getfield(ex) ?
+    (ir[v] = xcall(Zygote, :literal_getfield, ex.args[2], Val(unwrapquote(ex.args[3])))) :
     ex
```

Comment:
I think the multiline trinary is strictly harder to read than `if-else`
but morei mportantly we should be consistent withthe `literal_getproperty` above

Reply:
I agree. I just tried to keep consistency with the code base, but I think I can make an exception here